lib/xctest.js

```diff
@@ -0,0 +1,180 @@
+import { startHouseArrestService, startTestmanagerdService, startInstallationProxyService, startInstrumentService } from './services';
+import { INSTRUMENT_CHANNEL } from './instrument';
+import { TESTMANAGERD_CHANNEL } from './testmanagerd';
+import { DTXMessageAuxBuffer } from './instrument/headers';
+import { NSUUID, XCTestConfiguration } from './instrument/transformer/nskeyed';
+import { uuidV4 } from '@appium/support/build/lib/util';
+import { getOSVersion } from './utilities';
+import log from './logger';
+import B from 'bluebird';
+
+class Xctest {
+    constructor(udid, xctestBundleId, targetBundleId = undefined) {
+        this.udid = udid;
+        this.running = false;
+        this._executing = false;
+        this.xctestBundleId = xctestBundleId;
+        this.targetBundleId = targetBundleId;
+    }
+
+    async _launchAppRunner(majorVersion, sessionIdentifier) {
+        const { PROCESS_CONTROL } = INSTRUMENT_CHANNEL;
+        const installation = await startInstallationProxyService(this.udid);
+        const lookupResult = await installation.lookupApplications({ bundleIds: [this.xctestBundleId] });
+        installation.close();
+        const appInfo = lookupResult[this.xctestBundleId];
+        if (!appInfo) {
+            log.error(`${this.xctestBundleId} not found!`);
+            return;
+        }
+        const signerIdentifier = appInfo.SignerIdentity;
+        log.debug('SignerIdentifier:', signerIdentifier);
+        const appContainer = appInfo.Container;
+        const execName = appInfo.CFBundleExecutable;
+        if (!execName.endsWith('-Runner')) {
```

Comment:
WDA specific? I wondered if we could remove this to make this `xctest`class more general. (or tweak the class name to indicate it is for WDA)

Reply:
All xctest should follow this rule, including WDA

Reply:
please then move the magic suffix into a constant, for example `XCSUITE_SUFFIX = '-Runner'`